service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -185,14 +185,32 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 		return nil
 	}
 
-	switch v.Interface().(type) {
-	case []byte:
+	if v.Kind() != reflect.Slice {
```

Comment:
At this point the code doesn't know the destination `reflect.Value` is a `[]byte` or not yet. The only thing that is known is the source value is a `dynamodb.AttributeValue.B` (aka []byte). We the code below needs to figure out how to copy the source byte slice into the destination's value.